src/java/org/apache/cassandra/db/memtable/TrieMemtable.java

```diff
@@ -246,15 +246,22 @@ public long put(PartitionUpdate update, UpdateTransaction indexer, OpOrder.Group
         MemtableShard shard = shards[boundaries.getShardForKey(key)];
         long colUpdateTimeDelta = shard.put(update, indexer, opGroup);
 
-        if (shard.data.reachedAllocatedSizeThreshold() && !switchRequested.getAndSet(true))
-        {
-            logger.info("Scheduling flush due to trie size limit reached.");
-            owner.signalFlushRequired(this, ColumnFamilyStore.FlushReason.MEMTABLE_LIMIT);
-        }
+        if (shard.data.reachedAllocatedSizeThreshold())
+            signalFlushRequired(ColumnFamilyStore.FlushReason.TRIE_LIMIT);
 
         return colUpdateTimeDelta;
     }
 
+    @Override
+    public void signalFlushRequired(ColumnFamilyStore.FlushReason flushReason)
+    {
+        if (!switchRequested.getAndSet(true))
```

Comment:
In theory the owner can decide to ignore some request types. If the first we issue happens to be one of the ignored, this is going to block other requests of different type.

The original point of this check was to avoid doing a switch request on every write once the trie limit has been reached; doing so is benign but involves synchronization and thus should happen a little less frequent than that. Other flush reasons (with the exception of the new vector count limit) wouldn't trigger as often and would be quite okay to pass to the owner as many times as triggered.

Reply:
So should we only filter out high frequency flush reasons? Or perhaps have a counter and use modular arithmetic to only send the periodically?

Reply:
> The original point of this check was to avoid doing a switch request on every write once the trie limit has been reached; doing so is benign but involves synchronization and thus should happen a little less frequent than that. Other flush reasons (with the exception of the new vector count limit) wouldn't trigger as often and would be quite okay to pass to the owner as many times as triggered.

For index count limit, we should also avoid signaling owner too often. In edd8cfb6ab1, added a param `skipIfSignaled` for `signalFlushRequired` to skip signaling if it has already requested to switch memtable.